src/GrommetMarker.jsx

```diff
@@ -1,21 +1,76 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import styled, { ThemeContext } from 'styled-components'
 import ReactDOMServer from 'react-dom/server';
 import {
   createElementObject,
   createPathComponent,
   extendContext,
 } from '@react-leaflet/core';
 import L from 'leaflet';
-import { Location } from 'grommet-icons';
+import { Box } from 'grommet';
+import { StatusCriticalSmall, StatusGoodSmall, StatusWarningSmall, StatusUnknownSmall } from 'grommet-icons';
+import { adjustPadStyle } from './utils/styles';
+
+const StyledBox = styled(Box)`
+  // to create the pin shape
+  border-bottom-right-radius: 0;
+  transform: rotate(45deg);
+  &:hover {
+    // TO DO revisit "hover" color with designers. Use of "background-layer-overlay" feels strange semantically
+    border: ${props => `${props.theme?.global?.borderSize?.[props.border.size] || props.border.size} solid ${props.theme?.global?.colors?.['background-layer-overlay']}`};
+  }
+`;
+
+const statuses = {
+  critical: {
+    color: 'status-critical',
+    icon: StatusCriticalSmall,
+    borderSize: '3px',
+  },
+  warning: {
+    color: 'status-warning',
+    icon: StatusWarningSmall,
+  },
+  good: {
+    color: 'status-ok',
+    icon: StatusGoodSmall,
+  },
+  unknown: {
+    color: 'status-unknown',
+    icon: StatusUnknownSmall
+  }
+}
+
+const StatusIcon = ({ status, theme }) => {
+  const border = { color: statuses[status].color, size: statuses[status].borderSize || 'small'};
+  const StatusIcon = statuses[status].icon;
+
+  return (
+    <StyledBox 
+      background="background-front"
+      border={border}
+      elevation="medium"
+      flex={false}
+      pad={adjustPadStyle(theme.global.edgeSize.xsmall, theme.global.borderSize[border.size] || border.size)}
+      round
+      theme={theme}
+      width={{min: 'fit-content'}}
+    >
+      {/* style needed to offset for rotation of location pin so triangle is vertical still */}
+      <StatusIcon color={statuses[status].color} size="small" style={status === 'warning' ? { transform: 'rotate(-45deg)' } : undefined} />
```

Comment:
Would using Box's background object be an alternative? `rotate` is a property on that object.

Do we lose anything if the icon is not a child?